int32 state
float64[6] current_joint
float64[16] current_T_matrix